### DISCUSSION TITLE
SAHI with latest MMDet/MMRotate/MMYOLO

### DISCUSSION THREAD

**Discussion by hhaAndroid** [obss/sahi#686] [Ideas]
Hi, Thank you so much for your wonderful  library.  I'm an MMDet/MMYOLO developer. We recently made an architecture upgrade and open-sourced MMEngine/MMYOLO. I would like to know if it is possible to collaborate further to support OpenMMLab 2.0 in the sahi repo.
Looking forward to your reply!

**Comment by fcakyon** [comment]
@hhaAndroid thanks a lot for contacting us!

We are open to such collaboration but I could not understand what changes should be done while updating from `mmdet==2.25.1 mmcv-full==1.6.1` to `OpenMMLab 2.0`. Would you mind guiding us?

**Reply by hhaAndroid** [reply]
We will provide a draft design. Do you have any contact information there?

**Reply by fcakyon** [reply]
You can mail me at fatihcagatayakyon[at]gmail[dot]com

**Reply by hhaAndroid** [reply]
OK

**Comment by fcakyon** [comment]
Furthermore, SAHI supports instance segmentation outputs, mmrotate support can be added to SAHI by handling mmrotate outputs as 4 points polygons.

**Comment by hhaAndroid** [comment]
@fcakyon Hi, we will start this project next month. Before we do that, I would like to ask if you have any technical documentation available? Or about the support process for mmdet in sahi?

For mmdet 3.x and mmyolo you can get some information from https://mmdetection.readthedocs.io/en/3.x/ and https://mmyolo.readthedocs.io/en/latest/.  We will also be releasing an English PR release soon.

**Reply by fcakyon** [reply]
Hello @hhaAndroid, we use high level api for mmdet integration.

init_detector:
https://github.com/obss/sahi/blob/2ddf8a8723ae04ef8976c656c346972a936a9773/sahi/models/mmdet.py#L28

inference_detector:
https://github.com/obss/sahi/blob/2ddf8a8723ae04ef8976c656c346972a936a9773/sahi/models/mmdet.py#L78

are there any changes in the usage of these `init_detector` and `inference_detector` functions?

**Reply by hhaAndroid** [reply]
The change is very small. I will try it in MMYOLO. I'll let you know once it's done

**Reply by thiagoribeirodamotta** [reply]
Was sahi ever added for training on MMDet?

**Reply by Rosal-1998** [reply]
i wonder can we use sahi in mmyolo now? thank you for your reply.
